src/pkg/projects/Microsoft.NETCore.DotNetHost/host.wxs

```diff
@@ -7,17 +7,36 @@
 
     <ComponentGroup Id="InstallSharedHostandDetectionKeys">
 
+      <!-- When installing the SharedHost; copy all files to the traditional default install location: 'ProgramFiles'\dotnet even when installing into a custom location.
+
+      User Scenerio:
+      When a user has installed a 2.1 SDK or similar SDK first so that the PATH environmental variable has 'ProgramFiles'\dotnet in a first position.
+      Then the user installs a 3.0 SDK thru VS to a different volume with VS so that 'VSSharedVolume'\dotnet is in a second position.
+      We want the 3.0 SharedHost to be on the PATH in the first position. Therefore, installing the SharedHost to both 'VSSharedVolume'\dotnet and
+      'ProgramFiles'\dotnet will achieve this.
+
+      On a 'clean' machine when installing into a custom location; the installation behavior is the same.
+      Rationale:
+        1. the legacy SDK can be subsequently installed
+        2. the user runs 'dotnet' commands directly against 'ProgramFiles'\dotnet\dotnet.exe -->
+
       <Component Id="cmpCoreHost" Directory="DOTNETHOME" Guid="{45399BBB-DDA5-4386-A2E9-618FB3C54A18}" >
-        <File Id="fileCoreHostExe" KeyPath="yes" Source="$(var.HostSrc)\dotnet.exe" />
+        <File Id="fileCoreHostExe" KeyPath="yes" Source="$(var.HostSrc)\dotnet.exe">
+          <CopyFile Id="copyFileCoreHostExe" DestinationDirectory="[$(var.Program_Files)]dotnet" />
```

Comment:
Failing to compile in PR validation:

> ...\host.wxs(25): error CNDL0014: The CopyFile/@DestinationDirectory attribute's value, '[ProgramFiles64Folder]dotnet', is not a legal identifier.  Identifiers may contain ASCII characters A-Z, a-z, digits, underscores (_), or periods (.).  Every identifier must begin with either a letter or an underscore.

I missed the importance of the rest of @joeloff's comment:

> You want this in the actual authoring after the preprocessor finishes:
> 
> `<Directory Id="ProgramFilesFolder">`
> `<CopyFile DestinationDirectory="[ProgramFilesFolder]">`
> 
> This will be the resolved value of ProgramFilesFolder

But that doesn't make sense with the error message above, is that meant to be something like this?

```xml
<Directory Id="ProgramFilesFolder">
  <CopyFile DestinationDirectory="dotnet">
```

Reply:
I'm going with my first non-elegant solution:          

```
<?if $(var.Platform)=x86?>
            <CopyFile Id="copyFileCoreHostExe" DestinationDirectory="[ProgramFilesFolder]dotnet" />
          <?elseif $(var.Platform)=x64?>
            <CopyFile Id="copyFileCoreHostExe" DestinationDirectory="[ProgramFiles64Folder]dotnet" />
          <?endif?>
```

Reply:
That looks like it'll get identical output as before once the preprocessor runs. 😕

Reply:
DestinationDirectory should be DestinationProperty

Set this value to a property that will have a value that resolves to the full path of the destination directory. The property does not have to exist in the installer database at creation time; it could be created at installation time by a custom action, on the command line, etc. This attribute cannot be specified in conjunction with DestinationDirectory.

Reply:
I needed to pre-define a 'ProgramFiles'\dotnet directory and then use that property in the CopyFiles. I've defined "PROGRAMFILES_DOTNET" conditionally in 'product.wxs' and then we can use it in 'host.wxs'